src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceTest.java

```diff
@@ -432,21 +416,16 @@ public void selectCandidate_validInputs_taskInProgress() {
 
         Task task = new Task();
         task.setId(1L);
+        task.setCandidates(new ArrayList<>(Collections.singletonList(helper)));
 
-        Application application = new Application();
-        application.setUser(helper);
-        application.setTask(task);
-
-        when(userRepository.findUserById(anyLong())).thenReturn(helper, creator);
-        when(taskRepository.findById(anyLong())).thenReturn(task);
-        when(applicationsRepository.findByUserAndTask(any(User.class), any(Task.class))).thenReturn(application);
-        when(userRepository.findUserByToken("validToken")).thenReturn(creator); // Ensure the token matches
+        when(userService.getUserById(anyLong())).thenReturn(helper, creator);
+        when(taskRepository.findById(anyLong())).thenReturn(Optional.of(task));
+        when(userService.getUserByToken("validToken")).thenReturn(creator);
 
         taskService.selectCandidate(taskPutDTO, "validToken");
 
         assertEquals(helper, task.getHelper());
         assertEquals(TaskStatus.IN_PROGRESS, task.getStatus());
```

Comment:
Task service is not a mock, which is why verify cannot be called on it. Generally, verify is only used to assert communication with other components, never with the component under test.